File: reversibles.py
# problem 145

import operator

def reverse_num(n):
	return int(str(n)[::-1])

#def odd_digits(n):
#	return reduce(operator.and_, map(lambda x: x % 2 == 1, map(int, list(str(n)))), True)

def odd_digits(n):
	odd = True
	while n:
		odd, n = odd & ((n%10)%2==1), n/10
	return odd

def main():
	limit = 1000000000
	print sum(1 for i in xrange(1, limit) if odd_digits(i + reverse_num(i)) and i%10 != 0)

if __name__ == '__main__':
	main()

File: reversibles.c
/* problem 145 */

#include <stdio.h>

#define false	(0)
#define true	(!false)

const int limit = 1000000000;

int reverse(int n) {
	int m = 0;
	while(n) {
		m = m*10 + n%10;
		n /= 10;
	}
	return m;
}

bool odd_digits(int n) {
	while (n) {
		if (n % 2 == 0) return false;
		n /= 10;
	}
	return true;
}


int main(void) {
	int i = 0;
	int count = 0;
	for (i = 0; i < limit; ++i) {
		if (i%10 == 0) continue;
		if (odd_digits(i + reverse(i))) count++;
	}
	printf("%d\n", count);
	return 0;
}